### PR TITLE
Added docs for CDC Transformations

### PR DESCRIPTION
**Pull request by kathancox**
Fixes [DOC-4799](https://cockroachlabs.atlassian.net/browse/DOC-4799), [DOC-4798](https://cockroachlabs.atlassian.net/browse/DOC-4798), [DOC-4780](https://cockroachlabs.atlassian.net/browse/DOC-4780), [DOC-5324](https://cockroachlabs.atlassian.net/browse/DOC-5324)

This PR adds the following for the new CDC transformations feature:

- Page dedicated to CDC transformations — description, syntax, example use cases, limitations
- References to expressions on the `CREATE CHANGEFEED` page
- Added tracked GH limitation to the general CDC limitations
- Referenced expressions on the Change Data Capture Overview page

### CODE REVIEW

**Comment by kathancox** [comment]
Here is the link to the rendered preview for the main Changefeed Expressions page: 
https://deploy-preview-15308--cockroachdb-docs.netlify.app/docs/dev/changefeed-expressions.html

File: v22.2/create-changefeed.md [hunk 1]
```diff
@@ -28,8 +34,20 @@ To create a changefeed, the user must be a member of the `admin` role or have th
...
+
+### Changefeed expression parameters
+
+The [Changefeed Expressions](changefeed-expressions.html) page provides detail on the functionality, syntax, and use cases for changefeeds created with expressions. 
+
+Parameter | Description
+----------|------------
+`sink` | The location of the configurable sink. The scheme of the URI indicates the type. For more information, see [Sink URI](#sink-uri).
+`option` / `value` | For a list of available options and their values, see [Options](#options).
+`target_list` | The columns to emit data from.
```
Comment (HonoreDB) [inline v22.2/create-changefeed.md:48]: This line looks a little off? In my head the syntax looks something like

`CREATE CHANGEFEED [WITH options] [INTO sink] AS SELECT projection FROM target [WHERE predicate]`

**Comment by kathancox** [comment]
Thanks Aaron, I will implement your feedback tomorrow! I will also need to update the naming from Changefeed Expressions to CDC Transformations as per @amruss

File: v22.2/changefeed-expressions.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+- [Aggregate](functions-and-operators.html#aggregate-functions) and [window functions](window-functions.html) (i.e., functions operating over many rows).
+
+## Changefeed functions
+
+You can use any function from the [Functions and Operators page](functions-and-operators.html) unless it is marked as ["Volatile"](functions-and-operators.html#function-volatility). The following functions are changefeed specific:
+
+Function                  | Description
+--------------------------+----------------------
+`cdc_is_delete() `        | Returns `true` if the event is a deletion event.
+`cdc_prev()`              | Returns a JSON representation of a row's previous state.
```
Comment (miretskiy) [inline v22.2/changefeed-expressions.md:63]: can we add a callout that this function is (very likely) going to change.

File: v22.2/changefeed-expressions.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+~~~
+
+With this statement, you can create a changefeed that will emit messages including only the `name` and `owner` column data:
+
+~~~ sql
+CREATE CHANGEFEED INTO "scheme://sink-URI" WITH updated, schema_change_policy = 'stop' AS SELECT name, owner;
+~~~
+
+~~~
+{"after":{"office_dogs":{"owner":{"string":"Ashley"},"name":{"string":"Roach"}}},"updated":{"string":"1662044737959974465.0000000000"}}
```
Comment (miretskiy) [inline v22.2/changefeed-expressions.md:95]: @HonoreDB did we merge "raw" envelope? If so, we should update this output

File: v22.2/create-changefeed.md (hunk 1 above)
Comment (kathancox) [reply v22.2/create-changefeed.md:48]: Hmmm, so I ran all of expressions as `CREATE CHANGEFEED INTO sink WITH options AS SELECT....` 

With the option clause first, I receive a syntax error.

File: v22.2/changefeed-expressions.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+The following are not permitted in a changefeed expression:
+
+- [Volatile functions](functions-and-operators.html#function-volatility)
+- Sub-select queries
+- [Aggregate](functions-and-operators.html#aggregate-functions) and [window functions](window-functions.html) (i.e., functions operating over many rows).
+
+## Changefeed functions
+
+You can use any function from the [Functions and Operators page](functions-and-operators.html) unless it is marked as ["Volatile"](functions-and-operators.html#function-volatility). The following functions are changefeed specific:
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v22.2/changefeed-expressions.md]: @HonoreDB I've included an additional point here that the functions also listed in the Limitations section above this one are also not included. Does that make sense? Or, are there any others that I'm missing?

File: v22.2/changefeed-expressions.md (hunk 2 above)
Comment (kathancox) [reply v22.2/changefeed-expressions.md]: I added a note box to the Changefeed Functions section here to say that these functions are in preview and subject to change. OK?

File: v22.2/cdc-transformations.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+The following are not permitted in a changefeed expression:
+
+- [Volatile functions](functions-and-operators.html#function-volatility)
+- Sub-select queries
+- [Aggregate](functions-and-operators.html#aggregate-functions) and [window functions](window-functions.html) (i.e., functions operating over many rows).
+
+## Changefeed functions
+
+{{site.data.alerts.callout_info}}
+The following functions are in preview and **subject to change** with continued development.
```
Comment (amruss) [inline v22.2/cdc-transformations.md:59]: I think maybe we should put this caveat on the syntax area as well? I know we've talked about modifying the syntax pre moving out of preview

File: v22.2/cdc-transformations.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+You can use any function from the [Functions and Operators page](functions-and-operators.html) unless it is marked as ["Volatile"](functions-and-operators.html#function-volatility) or is listed in [Limitations](#limitations). The following functions are changefeed specific:
+
+Function                  | Description
+--------------------------+----------------------
+`cdc_is_delete() `        | Returns `true` if the event is a deletion event.
+`cdc_prev()`              | Returns a JSON representation of a row's previous state. 
+`cdc_updated_timestamp()` | Returns the event's update timestamp. This is typically the MVCC timestamp, but can differ. For example, if the table is undergoing [schema changes](online-schema-changes.html).
+
+## Examples
```
Comment (amruss) [inline v22.2/cdc-transformations.md:70]: I think we should add a geofiltering section. This allows us to create a region-specific changefeed for regional by row tables. It is the same as filtering on a column, but the column is the crdb_region column which is a hidden column that exists for all regional by row tables

Comment (kathancox) [reply v22.2/cdc-transformations.md:70]: Oh sure yeah, that makes sense! I will work on adding that.

File: v22.2/cdc-transformations.md (hunk 5 above)
Comment (kathancox) [reply v22.2/cdc-transformations.md:59]: The whole page has the preview feature label at the top... having three or more notes on things being subject to change might be a bit too much? If we don't feel that the one at the top is enough, perhaps I could combine and make a second note box (or warning?) further down near both the syntax and functions that warns the functions and syntax are in development and subject to change. 

In theory, the feature is marked as Preview, which covers any changes. But, I understand someone could miss that and be frustrated especially with changing syntax.

File: v22.2/changefeed-expressions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (HonoreDB) [reply v22.2/changefeed-expressions.md]: I think the code still bans most functions marked as stable (for example, `array_to_string`). Supported functions are functions marked as Immutable plus the following:
	"jsonb_build_array",
	"jsonb_build_object",
	"to_json",
	"to_jsonb",
	"row_to_json",
	"overlaps",
	"pg_collation_for",
	"pg_typeof",
	"quote_literal",
	"quote_nullable",

and minus the limitations you already have.

File: v22.2/cdc-transformations.md (hunk 5 above)
Comment (amruss) [reply v22.2/cdc-transformations.md]: Hmm yeah that's fair, my guess is we'll still allow the old syntax anyways and just change the docs so maybe not needed

**Comment by kathancox** [comment]
@amruss @HonoreDB @miretskiy Would you mind taking another look over for further feedback?

**Review by HonoreDB — APPROVED** [review]

**Review by amruss — COMMENTED** [review]
One nit: In the "recover lost messages" section, you say "In the event that an incident downstream has affected some rows, you may need a way to recover or evaluate the specific rows. Create a new changefeed that only watches for the affected row(s):"

I would add a caveat here that this will only include row history for a row that has been changed since the gc.tll window. If the change occurred a longer time ago than the gc.ttl window, then we won't see all the changes. (Unless for this example, ride_id is not the primary key for the row... in which case we should clarify that)

**Review by amruss — APPROVED** [review]
Small nit comment but otherwise LGTM

**Comment by kathancox** [comment]
> One nit: In the "recover lost messages" section, you say "In the event that an incident downstream has affected some rows, you may need a way to recover or evaluate the specific rows. Create a new changefeed that only watches for the affected row(s):"
> 
> I would add a caveat here that this will only include row history for a row that has been changed since the gc.tll window. If the change occurred a longer time ago than the gc.ttl window, then we won't see all the changes. (Unless for this example, ride_id is not the primary key for the row... in which case we should clarify that)

Added!

File: v22.2/cdc-transformations.md [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+You can use any CockroachDB-supported SQL expression syntax that is not listed in [limitations](#limitations) to build a changefeed query.
+
+See the [Examples](#examples) section for further use cases.
+
+## Syntax
+
+There are two possible components to CDC transformations:
+
+- _Projections_ will select the columns that you want to emit data from. 
+- _Predicates_ will restrict the selected column change data based on the filters you apply.
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:27]: nit: these descriptions overlap just a little. Maybe if framed as input v output (i.e. modify the incoming data, then modify the outgoing data), something like:

> - _Projections_ limit the queried data to specific columns that you want to emit data from.
> - _Predicates_ limit the resulting column change data payload based on the filters you apply.

File: _includes/v22.2/known-limitations/cdc-transformations.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+- It is necessary to pass the [`schema_change_policy='stop'`](create-changefeed.html#schema-policy) option in the changefeed creation statement when using the {% if page.name == "cdc-transformations.md" %} CDC transformations {% else %} [CDC transformations](cdc-transformations.html){% endif %} format. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/85143).
```
Comment (andf-crl) [inline _includes/v22.2/known-limitations/cdc-transformations.md:1]: nit: Consider starting this sentence with `Currently, it is necessary ...` to indicate that an active PR to address this exists. Although ... that PR is closed!

File: _includes/v22.2/known-limitations/cdc-transformations.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+- You can only apply CDC transformations on a single table in each statement.
+- Some stable functions, notably functions that return MVCC timestamps, are overridden to return the MVCC timestamp of the event.
+- You cannot [alter](alter-changefeed.html) a changefeed that uses CDC transformations. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/83033)
```
Comment (andf-crl) [inline _includes/v22.2/known-limitations/cdc-transformations.md:4]: same nit: consider `Currently, ...`

File: _includes/v22.2/known-limitations/cdc-transformations.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+- It is necessary to pass the [`schema_change_policy='stop'`](create-changefeed.html#schema-policy) option in the changefeed creation statement when using the {% if page.name == "cdc-transformations.md" %} CDC transformations {% else %} [CDC transformations](cdc-transformations.html){% endif %} format. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/85143).
+- You can only apply CDC transformations on a single table in each statement.
+- Some stable functions, notably functions that return MVCC timestamps, are overridden to return the MVCC timestamp of the event.
```
Comment (andf-crl) [inline _includes/v22.2/known-limitations/cdc-transformations.md:3]: nit: we could link "stable functions" here: https://www.cockroachlabs.com/docs/stable/functions-and-operators.html#built-in-functions

File: _includes/v22.2/known-limitations/udf-cdc-transformations.md [hunk 11]
```diff
@@ -0,0 +1 @@
+- {% if page.name == "cdc-transformations.md" %} CDC transformations {% else %} [CDC transformations](cdc-transformations.html) {% endif %} do not support {% if page.name == "user-defined-functions.md" %} user-defined functions {% else %} [user-defined functions](user-defined-functions.html). {% endif %}
```
Comment (andf-crl) [inline _includes/v22.2/known-limitations/udf-cdc-transformations.md:1]: Clever!!

File: v22.2/cdc-transformations.md [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+
+## Limitations
+
+{% include {{ page.version.version }}/known-limitations/cdc-transformations.md %}
+
+## CDC transformation function support
+
+You can use the following functions in CDC transformation queries:
+
+- "Immutable" as marked on the [Functions and Operators page](functions-and-operators.html).
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:51]: nit: recommend starting with the noun in focus (this also matches phrasing under `### Unsupported functions`): 

> Functions marked as "Immutable" on the [Functions and Operators page](functions-and-operators.html)

File: v22.2/cdc-transformations.md [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+
+    {{site.data.alerts.callout_info}}
+    Changefeed functions are in preview and **subject to change** with continued development.
+    {{site.data.alerts.end}}
+
+    Function                  | Description
+    --------------------------+----------------------
+    `cdc_is_delete() `        | Returns `true` if the event is a deletion event.
+    `cdc_prev()`              | Returns a JSON representation of a row's previous state. 
+    `cdc_updated_timestamp()` | Returns the event's update timestamp. This is typically the MVCC timestamp, but can differ. For example, if the table is undergoing [schema changes](online-schema-changes.html).
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:62]: nit: sentence fragment could be merged in like so:

> This is typically the MVCC timestamp but can differ in some cases, such as when a table is undergoing [schema changes](online-schema-changes.html).

File: v22.2/cdc-transformations.md [hunk 14]
```diff
@@ -0,0 +1,311 @@
...
+  - `pg_typeof()`
+  - `quote_literal()`
+  - `quote_nullable()`
+
+### Unsupported functions
+
+You can **not** use the following functions with CDC transformations:
+
+- Functions marked as "Volatile" on the [Functions and Operators page](functions-and-operators.html).
+- Functions listed in [Limitations](#limitations) on this page.
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:81]: nit: recommend either `listed under the [Limitations] section ...` or `listed in the [Limitations] section ...`

File: v22.2/cdc-transformations.md [hunk 15]
```diff
@@ -0,0 +1,311 @@
...
+- Functions marked as "Stable" on the [Functions and Operators page](functions-and-operators.html), **except** for those listed previously.
+
+## Examples
+
+CDC transformations allow you to customize your changefeed for particular scenarios. This section outlines several possible use cases for CDC transformations. 
+
+See [`CREATE CHANGEFEED`](create-changefeed.html) for examples on using the foundational syntax to create a changefeed. For information on sinks, see the [Changefeed Sinks](changefeed-sinks.html) page.
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:88]: Nice, all the building blocks I need for the examples are carefully listed right here. Good stuff!

File: v22.2/cdc-transformations.md [hunk 16]
```diff
@@ -0,0 +1,311 @@
...
+Filtering delete messages from your changefeed is helpful for certain outbox table use cases. See [Transformations and the outbox pattern](#transformations-and-the-outbox-pattern) for further detail.
+
+### Geofilter a changefeed
+
+When you are working with a [`REGIONAL BY ROW` table](set-locality.html#regional-by-row), you can filter the changefeed on the `crdb_region` column to create a region-specific changefeed:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED INTO sink WITH schema_change_policy = 'stop' AS SELECT * FROM table WHERE crdb_region = 'europe-west2';
+~~~
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:129]: Wow, this is gonna be a wildly-popular usage of this new feature.

File: v22.2/cdc-transformations.md [hunk 17]
```diff
@@ -0,0 +1,311 @@
...
+
+For more detail on targeting `REGIONAL BY ROW` tables with changefeeds, see [Changefeeds in Multi-Region Deployments](changefeeds-in-multi-region-deployments.html).
+
+### Stabilize the changefeed message schema
+
+As changefeed messages emit from the database, message formats can vary as tables experience [schema changes](changefeed-messages.html#schema-changes). You can select columns with typecasting to prevent message fields from changing during a changefeed's lifecycle:
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:135]: nit: worth linking `typecasting` here: https://www.cockroachlabs.com/docs/stable/data-types.html#data-type-conversions-and-casts ?

File: v22.2/cdc-transformations.md [hunk 18]
```diff
@@ -0,0 +1,311 @@
...
+The final changefeed created:
+
+{% include_cached copy-clipboard.html %}
+~~~sql 
+CREATE CHANGEFEED INTO 'scheme://sink-URI' WITH schema_change_policy='stop' AS SELECT * FROM movr.vehicle_location_histories WHERE left(ride_id::string, 1) IN ('c','d','e','f');
+~~~
+
+### View recent changes to a row
+
+You can use CDC transformations as a tool for debugging or investigation from the SQL shell.
```
Comment (andf-crl) [inline v22.2/cdc-transformations.md:164]: nit: match case for debug and investigate: 

> You can use CDC transformations as a tool for debugging or investigating issues from within the SQL shell.

File: v22.2/create-changefeed.md [hunk 19]
```diff
@@ -13,6 +13,12 @@ The `CREATE CHANGEFEED` [statement](sql-statements.html) creates a new {{ site.d
 
 We recommend reading the [Changefeed Messages](changefeed-messages.html) page for detail on understanding how changefeeds emit messages and [Create and Configure Changefeeds](create-and-configure-changefeeds.html) for important usage considerations.
 
+The [examples](#examples) on this page provide the foundational syntax of the `CREATE CHANGEFEED` statement. For examples on more specific use cases with changefeeds see the following pages:
+
+- [Change Data Capture Transformations](cdc-transformations.html)
+- [Changefeeds on Tables with Column Families](changefeeds-on-tables-with-column-families.html)
+- [Export Data with Changefeeds](export-data-with-changefeeds.html)
+
```
Comment (andf-crl) [inline v22.2/create-changefeed.md:21]: This is a great addition here!

File: v22.2/create-changefeed.md [hunk 20]
```diff
@@ -28,8 +34,20 @@ To create a changefeed, the user must be a member of the `admin` role or have th
...
 ----------|------------
 `table_name` | The name of the table (or tables in a comma separated list) to create a changefeed for.<br><br>**Note:** Before creating a changefeed, consider the number of changefeeds versus the number of tables to include in a single changefeed. Each scenario can have an impact on total memory usage or changefeed performance. See [Create and Configure Changefeeds](create-and-configure-changefeeds.html) for more detail.
-`sink` | The location of the configurable sink. The scheme of the URI indicates the type. For more information, see [Sink URI](#sink-uri) below.
-`option` / `value` | For a list of available options and their values, see [Options](#options) below.
+`sink` | The location of the configurable sink. The scheme of the URI indicates the type. For more information, see [Sink URI](#sink-uri).
+`option` / `value` | For a list of available options and their values, see [Options](#options).
+
+### CDC transformation parameters
+
+The [Change Data Capture Transformations](cdc-transformations.html) page provides detail on the functionality, syntax, and use cases for changefeeds created with transformations.
```
Comment (andf-crl) [inline v22.2/create-changefeed.md:42]: nit: recommend a single sentence explaining what a CDC transformation is, before linking out to it. What you have from `cdc-transformations.md` would be fine!

> Change data capture transformations allow you to define the change data emitted to your sink when you create a changefeed. See the [Change Data Capture Transformations](cdc-transformations.html) page for detail on the functionality, syntax, and use cases for changefeeds created with transformations.

**Review by andf-crl — APPROVED** [review]
LGTM with just a few minor language suggestions. Nice one!

File: _includes/v22.2/known-limitations/cdc-transformations.md (hunk 9 above)
Comment (kathancox) [reply _includes/v22.2/known-limitations/cdc-transformations.md:4]: We don't want to add forward-looking references per style guide. I think this still applies to both of these. Yes, this should be fixed, but we can't guarantee that. Probably better to just state what is the current state of the feature.

**Pull request merged**